app/src/main/java/com/gazlaws/codeboard/CodeBoardIME.java

```diff
@@ -272,9 +281,14 @@ public void onKey(int primaryCode, int[] KeyCodes) {
 //                        }
                 }
                 if (ke != 0) {
+
                     Log.i(getClass().getSimpleName(), "onKey: keyEvent " + ke);
-                    ic.sendKeyEvent(new KeyEvent(0, 0, KeyEvent.ACTION_DOWN, ke, 0, meta));
-                    ic.sendKeyEvent(new KeyEvent(0, 0, KeyEvent.ACTION_UP, ke, 0, meta));
+
+                    /* Added by Dimitris Stoukogiannis (aka ShadyGeek) on 14/2/2023.
+                        The if statement was added in order to prevent the space button from having an action down event attached to it. */
+                    if (primaryCode != 32) { ic.sendKeyEvent (new KeyEvent(0, 0, KeyEvent.ACTION_DOWN, ke, 0, meta)); }
```

Comment:
@ShadyGeek Why don't we want the ACTION_DOWN for the space button?

Reply:
If I understood everything correctly, having an ACTION_DOWN KeyEvent attached to the spacebar would result in the action taking place once the users pressed the button, while we want first to identify what the users are trying to do and afterwards produce the right output. I tried working this having the ACTION_DOWN KeyEvent enabled and it didn't work, so I removed it. Also, once I removed the ACTION_UP KeyEvent the programm didn't produce the right output, that's why I did it this way.

Reply:
>Also, once I removed the ACTION_UP KeyEvent the programm didn't produce the right output

Very interesting - thank you for the clarity.

Reply:
Of course man.